nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -193,6 +195,7 @@ def __init__(
         ckpt_load_optimizer: bool = True,
         ckpt_save_optimizer: bool = True,
         ddp: Union[DDPLiteral, DistributedDataParallelConfig] = "megatron",
+        fsdp: bool = False,
```

Comment:
should we rename it `torch_fsdp` because we know we'll have `mcore_fsdp` eventually?

Reply:
Do you think we can make this arg a string? ['mcore', 'torch'] or None. Or expand this arg to `fsdp` defualts to false, and `use_torch_fsdp` defaults to false? So:
`fsdp=Ture`, `use_torch_fsdp=True` -> torch_fsdp
`fsdp=Ture` -> mcore_fsdp (to be supported)
`nothing` -> ddp

Just trying to make the interface less confusing?